packages/pn-pa-webapp/src/components/Statistics/FilterStatistics.tsx

```diff
@@ -31,111 +30,73 @@ import { setStatisticsFilter } from '../../redux/statistics/reducers';
     // the formik validations for dates (which control the enable status of the "filtra" button)
     // must coincide with the input field validations (which control the color of the frame around each field)
     startDate: yup.date().min(tenYearsAgo).max(today),
-    endDate: yup
-      .date()
-      .min(dateIsDefined(startDate) ? startDate : tenYearsAgo)
-      .max(today),
+    endDate: yup.date().min(yup.ref('startDate')).max(today),
     selected: yup
       .mixed<SelectedStatisticsFilterKeys>()
       .oneOf(Object.values(SelectedStatisticsFilter)),
   });
 
   const formik = useFormik({
-    initialValues: initialValues(filter),
+    initialValues: {
+      startDate: filter.startDate,
+      endDate: lastDate && filter.endDate.getTime() === today.getTime() ? lastDate : filter.endDate,
+      selected: filter.selected,
+    },
     validationSchema,
+    enableReinitialize: true,
     /** onSubmit populates filter */
     onSubmit: () => {},
   });
 
-  const getRangeDates = (
-    range: SelectedStatisticsFilterKeys,
-    loading: boolean = false
-  ): [Date, Date] => {
+  const getRangeDates = (range: SelectedStatisticsFilterKeys): [Date, Date] => {
     switch (range) {
       case SelectedStatisticsFilter.lastMonth:
-        return [oneMonthAgo, today];
+        return [oneMonthAgo, lastDate ?? today];
       case SelectedStatisticsFilter.last3Months:
-        return [threeMonthsAgo, today];
+        return [threeMonthsAgo, lastDate ?? today];
       case SelectedStatisticsFilter.last6Months:
-        return [sixMonthsAgo, today];
+        return [sixMonthsAgo, lastDate ?? today];
       case SelectedStatisticsFilter.last12Months:
-        return [twelveMonthsAgo, today];
+        return [twelveMonthsAgo, lastDate ?? today];
       case SelectedStatisticsFilter.custom:
-        return loading
-          ? [filter?.startDate ?? defaultValues.startDate, filter?.endDate ?? defaultValues.endDate]
-          : [startDate ?? formik.values.startDate, endDate ?? formik.values.endDate];
+        return [formik.values.startDate, formik.values.endDate];
       default:
         return [defaultValues.startDate, defaultValues.endDate];
     }
   };
 
-  const setFilter = (type: SelectedStatisticsFilterKeys, loading = false): void => {
-    const [startDate, endDate] = getRangeDates(type, loading);
-    formik
-      .setValues({
-        startDate,
-        endDate,
-        selected: type,
-      })
-      .catch((error) => console.log(`${error}`));
-    setStartDate(startDate);
-    setEndDate(endDate);
-  };
-
   const handleSelectFilter = (type: SelectedStatisticsFilterKeys) => {
     const [startDate, endDate] = getRangeDates(type);
-    dispatch(
-      setStatisticsFilter({
-        startDate,
-        endDate,
-        selected: type,
-      })
-    );
+    dispatch(setStatisticsFilter({ startDate, endDate, selected: type }));
+    void formik.setValues({ ...formik.values, startDate, endDate, selected: type });
```

Comment:
move this before the dispatch and remove the ...formik.values because we are overriding all the values